### PR TITLE
Play 2.6 upgrade

### PR DESCRIPTION
**Pull request by TheDanielDoyle**

### CODE REVIEW

File: app/uk/gov/hmrc/apprenticeshiplevy/http/CustomErrorHandler.scala [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      statusCode match {
+        case NOT_FOUND =>
+          auditConnector.sendEvent(
+            dataEvent(
+              eventType       = "ResourceNotFound",
+              transactionName = "Resource Endpoint Not Found",
+              request         = request,
+              detail          = Map.empty
+            )
+          )
```
Comment (pascal22p) [inline app/uk/gov/hmrc/apprenticeshiplevy/http/CustomErrorHandler.scala:53]: Was it the case before? It's surprising a status code 404 would trigger some auditing.

File: test/uk/gov/hmrc/apprenticeshiplevy/connectors/DesConnectorSpec.scala [hunk 2]
```diff
@@ -20,21 +20,39 @@ import org.joda.time.{LocalDate, LocalDateTime}
...
-class DesConnectorSpec extends UnitSpec with MockitoSugar {
+class DesConnectorSpec extends UnitSpec with MockitoSugar with BeforeAndAfterEach{
+
+  val mockAppContext = mock[AppContext]
+  val mockHttp = mock[HttpClient]
+
+  def connector: DesConnector = new DesConnector(){
+    override def appContext: AppContext = mockAppContext
+    def baseUrl: String = "http://a.guide.to.nowhere"
+    def httpClient: HttpClient = mockHttp
```
Comment (pascal22p) [inline test/uk/gov/hmrc/apprenticeshiplevy/connectors/DesConnectorSpec.scala:47]: TODO: change this to wiremock

File: app/uk/gov/hmrc/apprenticeshiplevy/http/CustomErrorHandler.scala (hunk 1 above)
Comment (TheDanielDoyle) [reply app/uk/gov/hmrc/apprenticeshiplevy/http/CustomErrorHandler.scala:53]: Code taken directly from bootstrap.

File: app/uk/gov/hmrc/apprenticeshiplevy/http/CustomErrorHandler.scala [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+import play.api.mvc.{RequestHeader, Result}
+import uk.gov.hmrc.http._
+import uk.gov.hmrc.play.audit.http.connector.AuditConnector
+import uk.gov.hmrc.play.bootstrap.backend.http.{ErrorResponse, JsonErrorHandler}
+import uk.gov.hmrc.play.bootstrap.config.HttpAuditEvent
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class CustomErrorHandler @Inject()(
```
Comment (kurtlogan) [inline app/uk/gov/hmrc/apprenticeshiplevy/http/CustomErrorHandler.scala:32]: :+1: Don't like to do it this way but not much choice atm, I'll message platops about the potential issue

File: conf/application.conf [hunk 4]
```diff
@@ -67,121 +87,12 @@ http.router = prod.Routes
...
-
-play.http.requestHandler = "play.api.http.GlobalSettingsHttpRequestHandler"
-
 metrics {
   name = ${appName}
   rateUnit = SECONDS
   durationUnit = SECONDS
   showSamples = true
-  jvm = true
+  jvm = false //TODO turn on in app-config-base
```
Comment (kurtlogan) [inline conf/application.conf:95]: Has this been done?

File: test/uk/gov/hmrc/apprenticeshiplevy/connectors/DesConnectorSpec.scala (hunk 2 above)
Comment (kurtlogan) [reply test/uk/gov/hmrc/apprenticeshiplevy/connectors/DesConnectorSpec.scala:47]: This is part of the post 2.6 work now

**Review by kurtlogan — APPROVED** [review]
Can merge - comment is to check app-config-base

**Pull request merged**